homeassistant/components/homekit_controller.py

```diff
@@ -0,0 +1,177 @@
+"""
+Support for Homekit device discovery.
+
+For more details about this component, please refer to the documentation at
+https://home-assistant.io/components/homekit_controller/
+"""
+import http
+import json
+import logging
+import os
+import uuid
+
+from homeassistant.components.discovery import SERVICE_HOMEKIT
+from homeassistant.config import get_default_config_dir
+from homeassistant.helpers import discovery
+from homeassistant.helpers.entity import Entity
+
+REQUIREMENTS = ['homekit==0.5']
+
+homekit_dir = 'homekit'
+DOMAIN = 'homekit_controller'
+
+_CONFIGURING = {}
+
+# Mapping from Homekit type to component.
+HOMEKIT_ACCESSORY_DISPATCH = {
+    'lightbulb': 'light',
+}
+
+KNOWN_ACCESSORIES = []
+KNOWN_DEVICES = {}
+DEVICES = []
+
+_LOGGER = logging.getLogger(__name__)
+
+
+def homekit_http_send(self, message_body=None):
+    """Send the currently buffered request and clear the buffer.
+
+    Appends an extra \\r\\n to the buffer.
+    A message_body may be specified, to be appended to the request.
+    """
+    self._buffer.extend((b"", b""))
+    msg = b"\r\n".join(self._buffer)
+    del self._buffer[:]
+
+    if message_body is not None:
+        msg = msg + message_body
+
+    self.send(msg)
+
+
+# pylint: disable=unused-argument, too-many-function-args
+def setup(hass, config):
+    """Set up for Homekit devices."""
+    import homekit
+
+    _LOGGER.info("Setting up homekit")
+
+    class hk_device(Entity):
```

Comment:
Any reason you define this inside the `setup` method and not outside of it?

Reply:
Fixed